src/features/api-keys/lib.ts

```diff
@@ -107,7 +127,22 @@ const getKeyMetadata = (reqBody: { name?: any; description?: any }) => {
 		);
 	}
 
-	return { name, description };
+	if (
+		expiryDate != null &&
+		(typeof expiryDate !== 'string' || isNaN(Date.parse(expiryDate)))
+	) {
+		throw new errors.BadRequestError(
+			'Key expiry date should be a value formatted as YYYY-MM-DD',
```

Comment:
Why are we having the expiry as `YYYY-MM-DD`? We don't specify dates in that format anywhere else in the API that I know of?

Reply:
Updated based on input [here](https://www.flowdock.com/app/rulemotion/resin-tech/threads/vx-rBrpDBqXtkdtMmSU5MmPYSlb).